### ISSUE TITLE
bug - LightCurveCollection index by slice returns a list rather than LightCurveCollection 

### ISSUE THREAD

**Issue by orionlee** [lightkurve/lightkurve#1069]
#### Problem description

- When indexing a `LightCurveCollection` by slice, e.g., `lcc[1:5]`, it returns a `list` rather than `LightCurveCollection`.
- 'TargetPixelFileCollection` has similar problems.

#### Example

```python
import lightkurve as lk

sr = lk.search_lightcurve("TIC74482749", mission="TESS")
lcc = sr.download_all(download_dir="data")

# it works with boolean indexing
lcc[lcc.sector > 26].stitch().scatter()

# fails, because lcc[1:5] is a list 
lcc[1:5].stitch().scatter()
```

#### Environment

-  lightkurve version (e.g. 1.0b6): 2.0.9

**Comment by barentsen** [comment]
`lcc[1:5]` returning a list instead of a collection is confusing for sure.  I agree this is a bug!

**Comment by barentsen** [comment]
Thanks to @orionlee in #1077, the bugfix for this issue will be included in Lightkurve v2.0.10!
